bin/convert-argv.js

```diff
@@ -40,7 +40,9 @@ module.exports = function(...args) {
 		argv["output-filename"] = path.basename(output);
 		argv["output-path"] = path.dirname(output);
 	}
-
+	if(!argv.mode) {
```

Comment:
can't this be set as `default: 'none'` in the [options](https://github.com/webpack/webpack-cli/blob/master/bin/config-yargs.js#L90-L95)? Same effect.

Reply:
The two statements above are relying on it not to be set, so that's why it's placed here

Reply:
The same does not happen when you set `default: none` via `yargs`? If not set, it gets the default value? (just curiosity btw)

Reply:
Yup. I think @PlayMa256 is right. Config phase happens first.

Reply:
Okay gotcha, will rebase the tomorrow, internet here is pretty bad

Reply:
Actually I don't think we should set default as none, webpack will warn us, that's intuitive and it informs the user about the choices they're making